src/main/java/randoop/generation/Detective.java

```diff
@@ -0,0 +1,311 @@
+package randoop.generation;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Executable;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Queue;
+import java.util.Set;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.plumelib.util.CollectionsPlume;
+import randoop.DummyVisitor;
+import randoop.ExecutionOutcome;
+import randoop.NormalExecution;
+import randoop.operation.CallableOperation;
+import randoop.operation.ConstructorCall;
+import randoop.operation.MethodCall;
+import randoop.operation.TypedClassOperation;
+import randoop.operation.TypedOperation;
+import randoop.sequence.ExecutableSequence;
+import randoop.sequence.Sequence;
+import randoop.test.DummyCheckGenerator;
+import randoop.types.NonParameterizedType;
+import randoop.types.Type;
+import randoop.types.TypeTuple;
+import randoop.util.ListOfLists;
+import randoop.util.Randomness;
+import randoop.util.SimpleList;
+
+/**
+ * A demand-driven approach to construct inputs. Randoop works by selecting a method, then trying to
+ * find inputs to that method. Ordinarily, Randoop works bottom-up: if Randoop cannot find inputs
+ * for the selected method, it gives up and selects a different method. This demand-driven approach
+ * works top-down: if Randoop cannot find inputs for the selected method, then it looks for methods
+ * that create values of the necessary type, and recursively tries to call them.
+ *
+ * <p>The demand-driven approach implements the "Detective" component described by the paper "GRT:
+ * Program-Analysis-Guided Random Testing" by Ma et. al (appears in ASE 2015):
+ * https://people.kth.se/~artho/papers/lei-ase2015.pdf .
+ */
+public class Detective {
```

Comment:
The name "Detective" is not at all evocative of what this feature does.  (This is not your fault!)  I suggest that you change the name and documentation to focus on "demand-driven".  Please mention that it implements Detective from the GRT paper as a (useful!) side note, but not as the main documentation.

Reply:
Renamed to "DemandDrivenInputCreation" for the most part.